File: level2/ex14.js
function solution(w, h) {
  let count = 0;

  // 직사각형 대각선을 나타내는 일차 함수
  // f(x) = (h / w) * x 인데
  // h / w 계산에서 소수점이 부정확할 수 있어
  // h * x 먼저하고 w로 나눔
  const fx = (x) => h * x / w;

  for (let i = 0; i < w; i++) {
    const prev = Math.floor(fx(i));
    const next = Math.ceil(fx(i + 1));
    const diff = next - prev;

    count += diff;
  }

  return w * h - count;
}

File: level2/ex14.py
from math import gcd, ceil, floor

def solution(w, h):
    if w==1 or h==1:
        return 0
    # x값 0부터 a까지 대각선이 지나는 사각형의 개수 계산
    n = gcd(w,h)
    a = int(w/n)
    b = int(h/n)
    f = lambda x: -b/a*x+b
    g = lambda x: ceil(f(x))-floor(f(x+1))
    unused = sum([g(i) for i in range(a)])
    # 전체 사각형의 개수 - 대각선이 지나는 전체 사각형의 개수
    return w*h-unused*(w/a)